devnet-sdk/system2/chain.go

```diff
@@ -0,0 +1,56 @@
+package system2
+
+import (
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+)
+
+// Chain is an ethereum chain, common between L1 and L2.
+// For L1 or L2 specifics, see L1Chain and L2Chain extensions.
+// A chain hosts configuration resources and tracks participating nodes.
+type Chain interface {
+	Common
+
+	ChainID() eth.ChainID
+
+	ChainConfig() *params.ChainConfig
+
+	Faucet() Faucet
+}
+
+type ChainConfig struct {
+	CommonConfig
+	ChainCfg *params.ChainConfig
```

Comment:
Tiny thing (and I am not sure whether to address it on just naming level or deeper):

We are directly exposing `go-ethereum` types here (and above in `ChainID()`). I would rather have them e.g. "namespace" (`EthChainID`, `EthChainConfig`) than sometimes echoing the names `ChainID`, sometimes abbreviating `ChainCfg`.

The deeper thing is of course to remove the tight coupling altogether by describing the `go-ethereum` structs as interfaces.

Reply:
Undoing the deep coupling requires more tools in the monorepo to work with a common interface instead of the `*params.ChainConfig`. I'll open an issue for that, I think it's out of scope for this PR.

Reply:
https://github.com/ethereum-optimism/optimism/issues/14959

Reply:
That said, will fix the type naming, as suggested in the other comment

Reply:
I'm renaming it to `Network` (= chain with its offchain resources). `system2.L2NetworkID{Key: "dev0", ChainID: eth.ChainIDFromUint64(123)}` seems more natural. Chain and network naming is distinct enough to not mix up devnet-SDK and ge-ethereum protocol types.